config/initializers/omniauth.rb

```diff
@@ -1,5 +1,46 @@
+# Required for SAML SSO - ensures consistent callback URLs and SP entity ID across environments
+# SAML authentication is sensitive to URL mismatches, so OmniAuth needs the correct host
+OmniAuth.config.full_host = ENV.fetch('FRONTEND_URL', 'http://localhost:3000')
+
 Rails.application.config.middleware.use OmniAuth::Builder do
   provider :google_oauth2, ENV.fetch('GOOGLE_OAUTH_CLIENT_ID', nil), ENV.fetch('GOOGLE_OAUTH_CLIENT_SECRET', nil), {
     provider_ignores_state: true
   }
+
+  # SAML provider with setup phase for multi-tenant configuration
```

Comment:
should we move this to enterprise ?

Reply:
@sojan done with [`e0a5a63` (#12319)](https://github.com/chatwoot/chatwoot/pull/12319/commits/e0a5a63f2e73d16c4a3374c11d571eb621670988)

PTAL

Reply:
@sojan-official done